base_business_document_import/models/business_document_import.py

```diff
@@ -0,0 +1,953 @@
+# Copyright 2015-2019 Akretion France
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api, _
+from odoo.tools import float_compare
+from odoo.addons.base_iban.models.res_partner_bank import validate_iban
+from odoo.exceptions import UserError
+from lxml import etree
+from io import BytesIO
+import mimetypes
+from urllib.parse import urlparse
+import logging
+logger = logging.getLogger(__name__)
+
+try:
+    import PyPDF2
+except ImportError:
+    logger.debug('Cannot import PyPDF2')
+
+
+class BusinessDocumentImport(models.AbstractModel):
+    _name = 'business.document.import'
+    _description = 'Common methods to import business documents'
+
+    @api.model
+    def user_error_wrap(self, error_msg):
+        assert error_msg
+        prefix = self._context.get('error_prefix')
+        if prefix and isinstance(prefix, str):
+            error_msg = '%s\n%s' % (prefix, error_msg)
+        raise UserError(error_msg)
+
+    @api.model
+    def _strip_cleanup_dict(self, match_dict):
+        if match_dict:
+            for key, value in match_dict.items():
+                if value and isinstance(value, str):
+                    match_dict[key] = value.strip()
+            if match_dict.get('country_code'):
+                match_dict['country_code'] = match_dict['country_code'].upper()
+            if match_dict.get('state_code'):
+                match_dict['state_code'] = match_dict['state_code'].upper()
+
+    @api.model
+    def _match_partner(
+            self, partner_dict, chatter_msg, partner_type='supplier'):
+        """Example:
+        partner_dict = {
+            'country_code': 'FR',
+            'state_code': False,
+            'vat': 'FR12448890432',
+            'email': 'roger.lemaire@akretion.com',
+            'website': 'www.akretion.com',
+            'name': 'Akretion France',
+            'ref': 'C1242',
+            'phone': '01.41.98.12.42',
+            }
+        The key 'phone' is used by the module
+        base_phone_business_document_import
+        """
+        rpo = self.env['res.partner']
```

Comment:
```suggestion
        partner = self.env['res.partner']
```

Reply:
For me a partner is an instance of the model 'res.partner' not the class represented by self.env['res.partner'] 
rpo is better because it's a shorcut. Just my opinion